compilers/concrete-compiler/compiler/lib/Bindings/Python/CompilerAPIModule.cpp

```diff
@@ -1081,6 +1080,28 @@ void mlir::concretelang::python::populateCompilerAPISubmodule(
       .doc() = "Definition of a partition (in terms of precision in bits and "
                "norm2 in value).";
 
+  // ------------------------------------------------------------------------------//
+  // EXTERNAL PARTITION //
+  // ------------------------------------------------------------------------------//
+  pybind11::class_<concrete_optimizer::utils::ExternalPartition>(
+      m, "ExternalPartition")
+      .def(init([](std::string name, uint64_t macroLog2PolynomialSize,
+                   uint64_t macroGlweDimension, uint64_t macroInternalDim,
+                   double maxVariance, double variance)
+                    -> concrete_optimizer::utils::ExternalPartition {
+             return concrete_optimizer::utils::ExternalPartition{
+                 name,
+                 macroLog2PolynomialSize,
+                 macroGlweDimension,
+                 macroInternalDim,
```

Comment:
not sure if macroInternalDim have any use now or in the future !?

Reply:
Not sure how this is used, but it is needed to construct the external partition structure in the optimizer ...